CMakeLists.txt

```diff
@@ -231,6 +240,18 @@ cprover_default_properties(
     xml)
 
 option(WITH_JBMC "Build the JBMC Java front-end" ON)
+
+# Don't build jbmc on MacOS running on ARM
+# See https://github.com/Homebrew/homebrew-core/blob/master/Formula/cbmc.rb
+# See https://github.com/diffblue/cbmc/issues/6343
+execute_process(COMMAND uname -p
```

Comment:
This I'm less happy about. Fundamentally, it looks like the issue is nothing to do with macos and/or M1 architecture. Its _entirely_ down to Java version. It just happens that _at this moment in time_, brew will install a JVM 17 preview release on M1 macs (but still installs a JVM16 on x86 macs). I strongly suspect that as soon as more JVM17's are installed on other platforms (and OpenJDK 17 has just been officially released: https://mail.openjdk.java.net/pipermail/jdk-dev/2021-September/006037.html) this will bite other platforms.